tests/Unit/DataStructures/Test_Identity.cpp

```diff
@@ -0,0 +1,45 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <array>
+#include <catch.hpp>
+
+#include "DataStructures/Tensor/Identity.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+template <size_t First, size_t Last>
+struct static_for {
+  template <typename Lambda>
+  static inline constexpr void apply(Lambda const& f) {
+    if (First < Last) {
+      f(std::integral_constant<size_t, First>{});
+      static_for<First + 1, Last>::apply(f);
+    }
+  }
+};
+template <size_t N>
+struct static_for<N, N> {
+  template <typename Lambda>
+  static inline constexpr void apply(Lambda const& /*f*/) {}
+};
+
+SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.Identity",
+                  "[DataStructures][Unit]") {
+  static_for<1, 4>::apply([&](auto dim) {
+      constexpr auto d = decltype(dim)::value;
+      const std::array<double, d> used_for_size = make_array<d>(0.0);
+      const auto identity_matrix{identity(used_for_size)};
+
+      static_for<0, d>::apply([&](auto i) {
```

Comment:
turn these into runtime loops with .get instead of get

Reply:
then take the whole body into 
```
namespace {
template <size_t Dim>
void test_identity(...);
}
```
and call it for each dim in the `SPECTRE_TEST_CASE`